src/backend/distributed/operations/stage_protocol.c

```diff
@@ -253,8 +252,8 @@ master_append_table_to_shard(PG_FUNCTION_ARGS)
 						errdetail("The underlying shard is not a regular table")));
 	}
 
-	char partitionMethod = PartitionMethod(relationId);
-	if (partitionMethod == DISTRIBUTE_BY_HASH || partitionMethod == DISTRIBUTE_BY_NONE)
+	if (IsCitusTableType(relationId, HASH_DISTRIBUTED) || IsCitusTableType(relationId,
```

Comment:
Same here, we don't need to support local tables `master_append_table_to_shard `, so should better be `NO_DIST_KEY`